cmd/entrypoint/secrets.go

```diff
@@ -313,6 +323,59 @@ func ReconcileSecrets(ctx context.Context, sh *SnapshotHolder) error {
 	return nil
 }
 
+// Returns secretName, secretNamespace from a provided (unstructured) filter if it contains a secret
+// Returns empty strings when the secret name and/or namespace could not be found
+func findFilterSecret(filter *unstructured.Unstructured, action func(snapshotTypes.SecretRef)) error {
+	// Just making extra sure this is actually a Filter
+	if filter.GetKind() != "Filter" && filter.GetKind() != "Filters" {
+		return errors.New("Non-Filter object in Snapshot.Filters")
```

Comment:
1. It would be good to use `fmt.Errorf` in order to include the kind in the error message
2. error messages start with a lower-case letter
3. why would it ever be `"Filters"` instead of `"Filter"`?